src/mac.c

```diff
@@ -0,0 +1,356 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+ * Copyright 2025 NXP
+ */
+
+#include "provider.h"
+#include <string.h>
+
+struct p11prov_mac_ctx {
+    P11PROV_CTX *provctx;
+
+    P11PROV_OBJ *key;
+    P11PROV_SESSION *session;
+
+    CK_MECHANISM_TYPE mech;
+    CK_MECHANISM_TYPE hash_mech;
```

Comment:
Sounds like hash_mech is only used to query the digst_map() when obtaining context params.

Instead of storing this type provide a helper function (or even just a macro) called hash_mech() that is just a case switch that returns the corresponding hash mechanism given an hmac mechanism.

Wherever you currently use provctx->hash_mech just call hash_mech(provctx->hmac_mech)